egcg_core/executor/cluster_executor.py

```diff
@@ -150,3 +168,7 @@ def _job_exit_code(self):
 
         self.info('Got %s states from %s jobs: %s', len(states), len(reports), states)
         return exit_status
+
+    def cancel_job(self):
+        msg = self._get_stdout('scancel ' + self.job_id)
```

Comment:
Don't we need to test _job_finished to avoid sending an scancel query to finished jobs ?